### ISSUE TITLE
Basic search for Place/Trace records

### ISSUE THREAD

**Issue by blms** [performant-software/itsb#10]
Original search: https://sameboats.org/search/

Search criteria:

- [x] Author names
- [x] City names
- [x] Notes
- [x] Citations

**Comment by rsimon** [comment]
We should define what the results of a search would look like in the UI.

- I could imagine this to be fairly straighforward for the __author names__. A successful search would simply switch on this author (in the trajectory or intersection view)?
- City names:
  - In trajectory mode, should the UI switch on all authors that go through this place? Should the current time filter be taken into account? (And zoom to the place)
  - In intersection mode: compute all intersections at this place (and zoom to the place)
- Notes and citations: I assume these would be tied to a place/author combination. We'd probably need a popup at the place to show a preview?
 
Also: should the search include an auto-complete dropdown? (Probably should, given that the search space is fairly limited, and users would be dealing with different types of search results.)

**Comment by blms** [comment]
@rsimon Yes, I haven't really thought through what a search would look like as a filter for the map visualizations. That may not actually be in scope, but if it would essentially already be accomplished in the course of reimplementing the existing search (i.e. as a separate page unrelated to visualization) then we might as well do it. However, let's not spend too much time on it, as it would be more of a "nice to have". The requirement as I see it would be to simply replicate the existing search.

That being said, I like your proposals, and I would say yes that the time filter should be taken into account as well.

> Notes and citations: I assume these would be tied to a place/author combination. We'd probably need a popup at the place to show a preview?

I've just been storing these in waypoints, in the `citations` and `relation.label` keys. In the [original CSVs](https://github.com/elotroalex/itsb/blob/dev4/python/in/movements/Aim%C3%A9%20C%C3%A9saire.csv), they're stored per waypoint. Can restructure if there would be a better way to represent them.

**Comment by rsimon** [comment]
@blms I was assuming the search should work directly in the map interface. But if we stick to re-implementing the existing search, then things should be relatively straightforward. I all works based on the waypoints, as you say.

- Author name brings up all waypoints for this author (as seems to be the case in the original search)
- Place name brings up all waypoints linked to that place
- Notes and citation matches waypoints directly

I can look into this. But we should probably sort out he React Router implementation first, since Search would then be another route, right?

**Comment by blms** [comment]
@rsimon Agreed! I'll open a PR for react-router before we start handling this.

**Comment by rsimon** [comment]
Just keeping a note here: we'll need a small refactoring, so that the Peripleo Store is available to all pages (see #17).

**Comment by rsimon** [comment]
I assigned myself to this. I think the only thing that's still missing is to display the formatted date interval for each waypoint in the search results, right?

**Comment by blms** [comment]
@rsimon Yes, I think so!
